charts/substra-backend/templates/add-incoming-nodes-operator.yaml

```diff
@@ -0,0 +1,70 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "substra.fullname" . }}-add-incoming-nodes
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app.kubernetes.io/name: {{ template "substra.name" . }}-add-incoming-nodes
+    app.kubernetes.io/part-of: {{ template "substra.name" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+        app.kubernetes.io/name: {{ template "substra.name" . }}-add-incoming-nodes
+        app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ template "substra.name" . }}-add-incoming-nodes
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: add-incoming-node
+        image: "{{ .Values.backend.image.repository }}:{{ .Values.backend.image.tag }}"
+        imagePullPolicy: "{{ .Values.backend.image.pullPolicy }}"
+        command: ["/bin/bash", "-c"]
+        args:
+        - |
+          while true; do
+          {{- range $index, $value := .Values.incomingNodes }}
+            ./manage.py create_incoming_node {{ .name }} {{ .secret }}
+          {{end}}
+            sleep 10
+          done
+        resources:
+          {{- toYaml .Values.resources | nindent 14 }}
+        env:
+          - name: DJANGO_SETTINGS_MODULE
+            value: backend.settings.{{ .Values.backend.settings }}
+          - name: BACKEND_DB_NAME
+            value: {{ .Values.postgresql.postgresqlDatabase }}
+          - name: BACKEND_DB_USER
+            value: {{ .Values.postgresql.postgresqlUsername }}
+          - name: BACKEND_DB_PWD
+            value: {{ .Values.postgresql.postgresqlPassword }}
+          - name: DATABASE_HOST
+            value: {{ .Release.Name }}-postgresql
+          - name: PYTHONUNBUFFERED
+            value: "1"
+      initContainers:
+      - name: wait-postgresql
+        image: jwilder/dockerize
+        command: ['dockerize', '-wait', 'tcp://{{ .Release.Name }}-postgresql:5432']
+      - name: init-migrate
+        image: "{{ .Values.backend.image.repository }}:{{ .Values.backend.image.tag }}"
+        command: ['python', 'manage.py', 'migrate']
+        env:
+          - name: DJANGO_SETTINGS_MODULE
+            value: backend.settings.{{ .Values.backend.settings }}
+          - name: BACKEND_ORG
+            value: {{ .Values.organization.name }}
+          - name: BACKEND_DB_NAME
+            value: {{ .Values.postgresql.postgresqlDatabase }}
+          - name: BACKEND_DB_USER
+            value: {{ .Values.postgresql.postgresqlUsername }}
+          - name: BACKEND_DB_PWD
+            value: {{ .Values.postgresql.postgresqlPassword }}
+          - name: DATABASE_HOST
+            value: {{ .Release.Name }}-postgresql
```

Comment:
I would suggest not using init-container here and do that in the loop. Using init-container, you only test that the database is reachable and up to date during the pod creation not at runtime